src/tactic/compute_degree.lean

```diff
@@ -176,12 +313,177 @@ do t ← target,
       success_if_fail $ gs.mfirst $ unify t) <|> fail "Goal did not change",
     try $ any_goals' norm_assum
 
+end interactive
+
+open interactive polynomial expr compute_degree
+/--
+`resolve_coeff` assumes that the goal has the form `f.coeff n = x`.  It is important that `x`
+is the result of applying `get_lead_coeff` to `f`!  Indeed, `resolve_coeff` reads through the
+expression making up `f` and matches each step with what `get_lead_coeff` would do in the current
+situation.  This means that all the side-goals that `resolve_coeff` leaves are always of the same
+shape `f'.coeff n' = get_lead_coeff c f'`.
+
+In some sense, this views `coeff _ <visible_top_degree>` and a "monad" converting between
+`R[X]` and `R`.  `resolve_coeff` performs the operations building `f` across the monad.
+-/
+meta def resolve_coeff : tactic unit := focus1 $ do
+t ← target >>= instantiate_mvars,
+`(coeff %%f %%n = _) ← whnf t reducible | fail!"{t} has the wrong form",
+match f with
+| `(@has_one.one %%RX %%_)            := refine ``(coeff_one_zero)
+| (app `(⇑C) _)                       := refine ``(coeff_C_zero)
+| (app `(⇑(@monomial %%R %%_ %%n)) x) := refine ``(coeff_monomial)
```

Comment:
`coeff_monomial` returns an if-expression but `get_lead_coeff` just returns the coefficient itself. I think here you should use a lemma of the form `coeff (monomial n x) n = x` instead.